src/shared/components/oncoprint/DataUtils.ts

```diff
@@ -102,8 +102,8 @@ export function fillGeneticTrackDatum(
     newDatum:Partial<GeneticTrackDatum>,
     hugoGeneSymbol:string,
     data:AnnotatedExtendedAlteration[]
-):GeneticTrackDatum {
-    newDatum.gene = hugoGeneSymbol;
+): Partial<GeneticTrackDatum> {
```

Comment:
no this is what we were talking about, just please leave as is, and leave the cast on line 166

Reply:
This return type correction is here to catch mismatched usage of the new `trackLabel` property in the various pieces of code and test that use the function. I already removed the complicated dependent type you objected to.

Reply:
Listing an incorrect return type of `GeneticTrackDatum` complicates that, as pretty much all code that references the function would have to disable type checking.

Reply:
I don't understand what you mean with the mismatched `trackLabel` usage?

Instead of changing the return type I'd rather change the argument type of newDatum to be more precise, because there's nothing imprecise about it - this function does return a complete GeneticTrackDatum

Reply:
Makes sense—that would make the usage clearer as well, as `makeGeneticTrackData()` could just use the return value without fighting against the type checker. I was going off the use-cases demonstrated in the tests, which all fed it an empty object.

Reply:
makes sense. thanks :+1:

Reply:
if you want, could you also change the argument type of `newDatum`?

Reply:
I'm not actually sure if that would be an improvement. On the one hand it would point out missing required inputs to the `fillGeneticTrackDatum()` function when using it, but on the other hand I don't know of a way to express it without making the function implementation fragile to any unrelated changes in the `GeneticTrackDatum` type. So for me, it's clear enough as-is now that I've adjusted the tests to document the intended use-case.

Reply:
okay, sounds good, thanks